### ISSUE TITLE
Negative qRel in HeatingCurve

### ISSUE THREAD

**Issue by rubenbaetens** [open-ideas/IDEAS#237]
Within ```IDEAS.Controls.HeatingCurves.HeatingCurve``` on state ```qRel``` as
```
qRel = IDEAS.Utilities.Math.Functions.smoothMax(x1=0, x2=(TRoo_in_internal - TOutOffSet)/(TRoo_nominal - TOutOffSet_nominal),deltaX=0.1);
```
in order to keep ```qRel``` positive to calculate ```qRel^(1/m)```. However, while using the component i get errors during my simulation stateing that ```qRel``` becomes negative (while ```minSup=true```) during the iteration, forcing my simulation to stop.

Did you experience the same, @Mathadon @arnoutaertgeerts ?

**Comment by Mathadon** [comment]
I have experienced this before as well. The problem is that Dymola integrates ‘too far’, and that it comes into a zone where the equation is no longer valid. A possible solution may be to make deltaX larger.

**Comment by rubenbaetens** [comment]
The integration problem is solved by stating ```smoothMax(x1=0.01, ...``` instead of ```smoothMax(x1=0, ...```. As long as this minimal ```x1``` is below the minimal modulation of your heat source, this cause no error at all.
